server/ce2e/plugin_test.go

```diff
@@ -0,0 +1,97 @@
+package ce2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams/clientmodels"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/testutils/containere2e"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+)
+
+func TestMessageHasBeenPostedNewMessageE2E(t *testing.T) {
+	t.Parallel()
+	mattermost, store, tearDown := containere2e.NewE2ETestPlugin(t)
+	defer tearDown()
+
+	client, err := mattermost.GetAdminClient(context.Background())
+	require.NoError(t, err)
+
+	user, _, err := client.GetMe(context.Background(), "")
+	require.NoError(t, err)
+
+	team, _, err := client.GetTeamByName(context.Background(), "test", "")
+	require.NoError(t, err)
+
+	channel, _, err := client.GetChannelByName(context.Background(), "town-square", team.Id, "")
+	require.NoError(t, err)
+
+	post := model.Post{
+		CreateAt:  model.GetMillis(),
+		UpdateAt:  model.GetMillis(),
+		UserId:    user.Id,
+		ChannelId: channel.Id,
+		Message:   "message",
+	}
+
+	err = store.SetUserInfo(user.Id, "ms-user-id", &oauth2.Token{})
+	require.NoError(t, err)
+
+	t.Run("Without Channel Link", func(t *testing.T) {
+		var newPost *model.Post
+		newPost, _, err = client.CreatePost(context.Background(), &post)
+		require.NoError(t, err)
+
+		time.Sleep(50 * time.Millisecond)
```

Comment:
These sleeps are an unfortunate consequence of using this method. I wonder if it would make sense to add a flag to be able to run the HasBeen hooks without a goroutine.

Reply:
I removed the other 2 sleeps, but this one, specifically is about something "not-happening", so I prefer to keep the Sleep here anyway. Having some flag to make the hooks Sync is adding more changes in the behavior between testing and real environment, and I prefer to avoid that, at least for now.